app/jobs/async_batch_update_job.rb

```diff
@@ -0,0 +1,32 @@
+class AsyncBatchUpdateJob < ActiveJob::Base
+  queue_as :batch_actions
+
+  BATCH_SIZE = 1
+
+  def perform(model_class, sql_query, changes)
+    total_count = model_class.constantize.count_by_sql(total_count_sql sql_query)
+    sql_query_reordered = order_by_id_sql sql_query
+
+    ((total_count.to_f / BATCH_SIZE).ceil).times do |batch_number|
```

Comment:
I think if instead of `sql_query` use `scoped_collection_records` (ActiveRecord::Relation), then in this method you can use:

```ruby
scoped_collection_records.order(id: :asc).find_each(batch_size: 1) { |r| r.update!(changes) }
```
without any calculation of ordered.

---

What do you think about this proposal, could it work?